infrastructure/repository/userservice/UserServiceSubjectRepository.go

```diff
@@ -0,0 +1,240 @@
+// Package userservice is for the userservice repository and related components
+package userservice
+
+import (
+	"authz/domain"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+const (
+	sortBy           = "principal"
+	defaultPageSize  = 20
+	defaultSortOrder = true
+
+	assumeNextPageAvailableByDefaultIfError = true // when retrieving a page of users and there is an error, should we still assume another page exists
+)
+
+// UserServiceSubjectRepository defines a repository that queries a user service using json requests of the type defined in userRepositoryRequest
+type UserServiceSubjectRepository struct {
+	URL        url.URL
+	HTTPClient http.Client
+	Paging     struct {
+		PageSize  int
+		SortOrder bool
+	}
+}
+
+// NewUserServiceSubjectRepository creates a new UserServiceSubjectRepository
+func NewUserServiceSubjectRepository(url url.URL, client http.Client) *UserServiceSubjectRepository {
+	return &UserServiceSubjectRepository{
+		URL:        url,
+		HTTPClient: client,
+		Paging: struct {
+			PageSize  int
+			SortOrder bool
+		}{PageSize: defaultPageSize, SortOrder: defaultSortOrder},
+	}
+}
+
+type userRepositoryRequest struct {
+	By struct {
+		AccountID  string `json:"accountId"`
+		WithPaging struct {
+			FirstResultIndex int    `json:"firstResultIndex"`
+			MaxResults       int    `json:"maxResults"`
+			SortBy           string `json:"sortBy"`
+			Ascending        bool   `json:"ascending"`
+		} `json:"withPaging"`
+	} `json:"by"`
+	Include struct {
+		AllOf []string `json:"allOf"`
+	} `json:"include"`
+}
+
+type userRepositoryResponse []struct {
+	ID     string `json:"id"`
+	Status string `json:"status"`
+}
+
+// GetByOrgID retrieves all members of the given organization
+func (u *UserServiceSubjectRepository) GetByOrgID(orgID string) (chan domain.Subject, chan error) {
+	subChan := make(chan domain.Subject)
+	errChan := make(chan error)
+
+	if !u.validateConfigAndOrg(orgID) {
+		errChan <- fmt.Errorf("UserServiceSubjectRepository config was not valid: %v", u)
```

Comment:
just a thought: maybe this validation is suited better at bootstrap package?!

Reply:
Maybe... I chose here, because it's closer to the point where a potentially heavy process kicks off that might be failing but still potentially taking significant time and resources. But maybe this fear is unfounded. I'm open-minded on this.

Reply:
Also, if we want to validate the orgId, that's a runtime thing that has to happen here.